src/binary_path.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+   Copyright (C) 2020 by the Battle for Wesnoth Project https://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+/**
+ * @file
+ * File-IO
+ */
+#define GETTEXT_DOMAIN "wesnoth-lib"
+
+#include "filesystem.hpp"
+#include "binary_path.hpp"
+
+#include "config.hpp"
+#include "deprecation.hpp"
+#include "game_config.hpp"
+#include "game_version.hpp"
+#include "gettext.hpp"
+#include "log.hpp"
+
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/filesystem/fstream.hpp>
+#include "game_config_view.hpp"
+
+
+static lg::log_domain log_filesystem("filesystem");
+#define DBG_FS LOG_STREAM(debug, log_filesystem)
+#define LOG_FS LOG_STREAM(info, log_filesystem)
+#define WRN_FS LOG_STREAM(warn, log_filesystem)
+#define ERR_FS LOG_STREAM(err, log_filesystem)
+
+namespace bfs = boost::filesystem;
+using boost::system::error_code;
+
+
+/**
+ *  The paths manager is responsible for recording the various paths
+ *  that binary files may be located at.
+ *  It should be passed a config object which holds binary path information.
+ *  This is in the format
+ *@verbatim
+ *    [binary_path]
+ *      path=<path>
+ *    [/binary_path]
+ *  Binaries will be searched for in [wesnoth-path]/data/<path>/images/
```

Comment:
This comment seems wrong; `images` should probably be replaced with `<kind>` or something similar, right?